libmamba/src/core/satisfiability_error.cpp

```diff
@@ -0,0 +1,348 @@
+                        break;
+                    }
+                    case SOLVER_RULE_PKG_REQUIRES:
+                    {
+                        if (!dep || !source)
+                        {
+                            warn_unexpected_problem(problem);
+                            break;
+                        }
+                        auto src_id
+                            = ensure_solvable(problem.source_id, std::move(source).value(), type);
+                        auto edge = RequireEdge{ DependencyInfo(dep.value()) };
+                        add_expanded_deps_edges(src_id, problem.dep_id, edge);
+                        break;
+                    }
+                    case SOLVER_RULE_JOB:
+                    case SOLVER_RULE_PKG:
+                    {
+                        if (!dep)
+                        {
+                            warn_unexpected_problem(problem);
+                            break;
+                        }
+                        // TODO check here that depId exists
+                        auto edge = RequireEdge{ DependencyInfo(dep.value()) };
+                        add_expanded_deps_edges(m_root_node, problem.dep_id, edge);
+                        break;
+                    }
+                    case SOLVER_RULE_JOB_NOTHING_PROVIDES_DEP:
+                    case SOLVER_RULE_PKG_NOTHING_PROVIDES_DEP:
+                    case SOLVER_RULE_JOB_UNKNOWN_PACKAGE:
+                    {
+                        if (!source || !dep)
+                        {
+                            warn_unexpected_problem(problem);
+                            break;
+                        }
+                        auto edge = RequireEdge{ DependencyInfo(dep.value()) };
+                        auto src_id
+                            = ensure_solvable(problem.source_id, std::move(source).value(), type);
+                        auto tgt_id
+                            = ensure_solvable(problem.target_id, std::move(dep).value(), type);
+                        m_graph.add_edge(src_id, tgt_id, std::move(edge));
+                    }
+                    case SOLVER_RULE_PKG_CONFLICTS:
+                    case SOLVER_RULE_PKG_SAME_NAME:
+                    {
+                        if (!source || !target)
+                        {
+                            warn_unexpected_problem(problem);
+                            break;
+                        }
+                        auto src_id
+                            = ensure_solvable(problem.source_id, std::move(source).value(), type);
+                        auto tgt_id
+                            = ensure_solvable(problem.target_id, std::move(target).value(), type);
+                        add_conflict(src_id, tgt_id);
+                        break;
+                    }
+                    default:
+                    {
+                        LOG_WARNING << "Problem type not implemented: " << problem.type;
```

Comment:
should we add these later: https://github.com/mamba-org/mamba/pull/1891/files#diff-75f7d674a04436c369336098a9a9f41fdf672796e9960b8a98d9414b0156feeeR118-R131 ?